src/cogent3/maths/distance_transform.py

```diff
@@ -1253,7 +1267,41 @@ def binary_dist_hamming(datamtx, strict=True):
     return dists
 
 
-def binary_dist_jaccard(datamtx, strict=True):
+def fast_jaccard(x: iter, y: iter):
+    xset = set(x)
+    yset = set(y)
+    return 1 - len(xset & yset) / len(xset | yset)
+
+
+@singledispatch
+def jaccard(x, y):
+    """Jaccard distance between two sets is a measure of similarity between two sets
+    (0) identical, (1) dissimilar"""
+    raise NotImplementedError(f"jaccard distance not implemented for {type(x)}")
+
+
+@jaccard.register
+def _(x: set, y: set):
+    if not x and not y:  # two empty sets are identical
+        return 0
```

Comment:
make sure all returns are floats, so change `0` to `0.0`. This will actually have an impact in Python 3.11 (when we migrate to it) where there's a new builtin performance optimisation capability

Reply:
ah yes.  good call